File: webapp/src/types/resume.ts
export interface PersonalInformation {
    name: string;
    email: string;
    phone: string;
    linkedin?: string;
}

export interface LatestEducation {
    degree: string;
    major: string;
    school: string;
    location: string;
    start_date: string;
    end_date: string;
}

export interface WorkExperience {
    title: string;
    company: string;
    location: string;
    duration: YOE;
    start_date: string;
    end_date: string;
    description: string;
}

export interface Project {
    name: string;
    description: string;
    start_date: string;
    end_date: string;
    link?: string[];
}

export interface YOE {
    year: number;
    month: number;
}

export interface Resume {
    personal_information: PersonalInformation;
    latest_education?: LatestEducation;
    work_experiences?: WorkExperience[];
    projects?: Project[];
    skills: string[];
    certifications?: string[];
    yoe: YOE;
}

File: resume-parser-core/core/workflows/models/resume.py
from typing import Optional

from pydantic import BaseModel


class PersonalInformation(BaseModel):
    name: str
    email: str
    phone: str
    linkedin: Optional[str]


class Education(BaseModel):
    degree: str
    major: str
    school: str
    location: str
    start_date: str
    end_date: str


class WorkExperience(BaseModel):
    title: str
    company: str
    location: str
    start_date: str
    end_date: str
    description: str


class Project(BaseModel):
    name: str
    description: str
    start_date: str
    end_date: str
    link: Optional[list[str]]


class Resume(BaseModel):
    personal_information: PersonalInformation
    latest_education: Education
    work_experiences: Optional[list[WorkExperience]]
    projects: Optional[list[Project]]
    skills: list[str]
    certifications: Optional[list[str]]
